src/core/server/http/http_service.ts

```diff
@@ -69,7 +73,8 @@ export class HttpService implements CoreService<HttpServiceSetup, HttpServiceSta
 
     const config = await this.config$.pipe(first()).toPromise();
 
-    return this.httpServer.setup(config);
+    const setup = this.httpServer.setup(config);
+    return { ...setup, ...{ createNewServer: (cfg: HttpConfig) => this.createServer(cfg) } };
```

Comment:
There is no need to spread here since you are defining on the object you are creating:

```ts
return { ...setup, createNewServer: (cfg: HttpConfig) => this.createServer(cfg) };
```

Reply:
Removing the spread here causes additional problems, that I'm not sure are worth the trade off.

Since `setup` is of type `HttpServerSetup` and this method needs to return `HttpServiceSetup`, I need do something to make typescript happy since `createNewServer` isn't part of `HttpServerSetup`, but rather `HttpServiceSetup`.

I can do the following:

```typescript
const setup = this.httpServer.setup(config) as HttpServiceSetup;
setup.createNewServer = (cfg: Partial<HttpConfig>) => this.createNewServer(cfg)
return setup
```

However, this runs into issues in the tests since `this.httpServer.setup()` returns `undefined`, and you can't add properties there:
```
  ● creates and sets up http server

    TypeError: Cannot set property 'createNewServer' of undefined

      78 |     const setup = this.httpServer.setup(config) as HttpServiceSetup;
      79 |     console.log('SETUP', setup);
    > 80 |     setup.createNewServer = (cfg: Partial<HttpConfig>) => this.createServer(cfg);
         |     ^
      81 |     return setup;
      82 |   }
      83 |

      at HttpService.setup (src/core/server/http/http_service.ts:80:5)
```

Which means I can do this:
```typescript
const setup = (this.httpServer.setup(config) || {}) as HttpServiceSetup;
setup.createNewServer = (cfg: Partial<HttpConfig>) => this.createNewServer(cfg)
return setup
```

To satisfy the tests, but I think:

```typescript
const httpSetup = this.httpServer.setup(config)
const setup = {...httpSetup, ...{createNewServer: (cfg: Partial<HttpConfig>) => this.createNewServer(cfg)}}
return setup
```

Might be the clearest approach?